packages/dds/tree/src/test/utils.ts

```diff
@@ -129,3 +149,28 @@ export class TestTreeProvider {
         });
     }
 }
+
+/**
+ * Run a custom "spy function" every time the given method is invoked.
+ * @param prototype - the prototype of the object which contains the method
+ * @param methodName - the name of the method
+ * @param spy - the spy function to run alongside the method
+ * @returns a function which will remove the spy function when invoked. Should be called exactly once
+ * after the spy is no longer needed.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function spyOnMethod(methodClass: Function, methodName: string, spy: () => void): () => void {
```

Comment:
I'm sure there some library somewhere that would offer this out of the box, and FF probably has some dependency on it already. I tend to prefer adding small utility functions for the sake of simplicity like you did here, but some people might object.
Feel free to ignore this comment.

Reply:
You're totally right. Jest, for example, is built on the idea of mocks/spies and certainly implements it better than me. IMO if we start using this as a general testing strategy in lots of our tests, not just in this one case (or a couple cases), we should invest the time to research a good library. I think this quick-and-dirty is okay for now. Soon we will write tests that actually summarize a SharedTree and validate data, and those will be testing the manual serialization functionality implicitly. So the test (and potentially this function) could even be removed them.